pkg/controller/elasticsearch/user/reconcile.go

```diff
@@ -122,12 +122,12 @@ func aggregateFileRealm(
 	}
 
 	// merge all file realms together, the last one having precedence
-	fileRealm := filerealm.MergedFrom(
-		internalUsers.fileRealm(),
-		elasticUser.fileRealm(),
-		associatedUsers.fileRealm(),
-		userProvidedFileRealm,
-	)
+	fileRealmUsers := []filerealm.Realm{internalUsers.fileRealm(), associatedUsers.fileRealm(), userProvidedFileRealm}
+	if !es.Spec.Auth.DisableElasticUser {
+		// not using append here as order matters.
+		fileRealmUsers = []filerealm.Realm{internalUsers.fileRealm(), elasticUser.fileRealm(), associatedUsers.fileRealm(), userProvidedFileRealm}
+	}
+	fileRealm := filerealm.MergedFrom(fileRealmUsers...)
```

Comment:
```suggestion
	fileRealm := filerealm.MergedFrom(
		internalUsers.fileRealm(),
		elasticUser.fileRealm(),
		associatedUsers.fileRealm(),
		userProvidedFileRealm,
	)
```

Reply:
No need for the conditional now that you have refactored the `reconcileElasticUser` method